test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -972,7 +972,7 @@ private static String groupName(ThreadGroup threadGroup) {
      * Returns a random subset of values (including a potential empty list)
      */
     public static <T> List<T> randomSubsetOf(Collection<T> collection) {
-        return randomSubsetOf(randomInt(Math.max(collection.size() - 1, 0)), collection);
+        return randomSubsetOf(randomInt(collection.size()), collection);
```

Comment:
I think this was an actual bug, where someone assumed randomInt to have an inclusive bound instead of an exclusive one (randomIntBetween for example uses inclusive bounds). I can add more javadoc here, but what this fix now actually implements is the definition of a subset.

Reply:
ignore my explanation about the bounds here. Still think it's a bug. I've enhanced javadocs in 1ce4fdc3839